homeassistant/util/ssl.py

```diff
@@ -164,3 +151,59 @@ def server_context_intermediate() -> ssl.SSLContext:
     context.set_ciphers(SSL_CIPHER_LISTS[SSLCipherList.INTERMEDIATE])
 
     return context
+
+
+def server_context_insecure() -> ssl.SSLContext:
+    """Return an SSL context for old SSL/TLS standards."""
+    context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
+
+    context.options |= (
+        ssl.OP_NO_SSLv2 | ssl.OP_NO_SSLv3 | ssl.OP_CIPHER_SERVER_PREFERENCE
+    )
+    if hasattr(ssl, "OP_NO_COMPRESSION"):
+        context.options |= ssl.OP_NO_COMPRESSION
+
```

Comment:
```suggestion
    context.options |= getattr(ssl, "OP_LEGACY_SERVER_CONNECT", 0x4)
```

I for insecure I think we really mean insecure